admin/digilan-admin.php

```diff
@@ -503,30 +512,37 @@ private static function update_user_form_fields()
         }
         $deleted_keys = array();
 
-        foreach ($_POST as $post_key => $post_value) {
-            // $post_key example :
-            // - dlt-custom-portal-field/display-name/gender/fr_FR
-            // - action
-            [$prefix, $field_option, $field_name, $lang_code] = explode('/', $post_key);
-            if ($prefix !== 'form-fields') {
-                continue;
-            }
-            if (in_array($field_name, $deleted_keys)) {
-                continue;
-            }
-            if ($field_option == 'delete') {
-                array_push($deleted_keys, $field_name);
-                unset($user_form_fields[$field_name]);
+        $form_languages = get_option('digilan_token_form_languages');
+
+        foreach($user_form_fields as $form_field_key=>$form_field_value) {
+            $safe_field_to_delete = DigilanTokenSanitize::sanitize_form_field_to_delete($_POST["form-fields/delete/$form_field_key"]);
+            if ($safe_field_to_delete === 'delete') {
+                unset($user_form_fields[$form_field_key]);
                 continue;
             }
 
-            $field_value = self::sanitize_form_field_option_by_type($field_option, $post_value);
-            if (false === $field_value) {
-                \DLT\Notices::addError(sprintf('Invalid %s for %s', $field_option, $field_name));
-                wp_redirect(self::getAdminUrl('form-settings'));
-                exit();
+            $field_type = $form_field_value['type'];
+            foreach($form_languages as $lang) {
+                if (false === $lang['implemented']) {
+                    continue;
+                }
+                $lang_code = $lang['code'];
+                $safe_display_name = self::sanitize_form_field_option_by_type('display-name', $_POST["form-fields/display-name/$form_field_key/$lang_code"]);
+                $user_form_fields[$form_field_key]['display-name'][$lang_code] = $safe_display_name;
+                
+                $safe_instruction = self::sanitize_form_field_option_by_type('instruction', $_POST["form-fields/instruction/$form_field_key/$lang_code"]);
+                $user_form_fields[$form_field_key]['instruction'][$lang_code] = $safe_instruction;
+                
+                if ($field_type === 'radio' || $field_type === 'select') {
+                    $safe_options = self::sanitize_form_field_option_by_type('options', $_POST["form-fields/options/$form_field_key/$lang_code"]);
+                    $user_form_fields[$form_field_key]['options'][$lang_code] = $safe_options;
+                }
+                
+                if ($field_type === 'number') {
+                    $safe_unit= self::sanitize_form_field_option_by_type('unit', $_POST["form-fields/unit/$form_field_key/$lang_code"]);
+                    $user_form_fields[$form_field_key]['unit'][$lang_code] = $safe_unit;
+                }
```

Comment:
all of that goes into $user_form_fields[$form_field_key] = routine

self::sanitize_form_field_option_by_type('display-name' cela sert a rien de fair eun switch case dedans apres tu factorise du vide

et lis ton post dedans

aussi un nom de fonction peut etre une variable...